msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AzureArcManagedIdentitySource.java

```diff
@@ -150,6 +150,16 @@ public ManagedIdentityResponse handleResponse(
         return super.handleResponse(parameters, response);
     }
 
+    private Optional<String> readChallengeFrom(IHttpResponse response) {
+        return response.headers()
+                .entrySet()
+                .stream()
+                .filter(entry -> WWW_AUTHENTICATE_HEADER.equalsIgnoreCase(entry.getKey()))
```

Comment:
Hi @bgavrilMS ! I extended the existing tests to run with both variations of the header key. Slipped in some small refactor as well to avoid duplication, hope you don't mind.